### PR TITLE
Bugfix: pagination page prop and ellipsis click

### PR DESCRIPTION
**Pull request by bshelkhonov**
Problem: 
1. pagination doesn't update on page prop change
2. when I click on ellipsis, pagination breaks because page number becomes 0 

Also I processed this file with prettier.

Basically the changes are:
1. add useeffect that tracks pageProp changes
2. check if value !== 0 before setting page
image

### CODE REVIEW

**Comment by mainsmirnov** [comment]
1. Eslint is failed for this PR
2. Check out controlled hooks `src/hooks/useEnsureControl.ts`, if you want to fix the problem you should use one of them

**Comment by bshelkhonov** [comment]
Prettier config was broken because of mistake in filename and also there was the conflict in rule `trailingComma` with eslint. I fixed it. 

Also I used `useCustomEnsuredControl` hook instead of `useState + useEffect`

File: src/components/Navigation/Pagination/hooks/usePagination.ts [hunk 1]
```diff
@@ -21,10 +20,13 @@ export const usePagination = ({
...
 }: UsePaginationProps): UsePaginationItem[] => {
-  const [page, setPageState] = useState(pageProp === undefined ? defaultPage : pageProp);
+  const [page, setPageState] = useCustomEnsuredControl({
+    value: pageProp,
+    defaultValue: defaultPage,
+  });
 
   const handleClick: UsePaginationProps['onChange'] = (event, value) => {
-    if (!pageProp) {
+    if (!pageProp && value !== 0) {
```
Comment (mainsmirnov) [inline src/components/Navigation/Pagination/hooks/usePagination.ts:29]: Can you explain when it's zero here?

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/hooks/usePagination.ts:29]: ellipsis click is treated as zero

https://github.com/Telegram-Mini-Apps/TelegramUI/blob/6a3882a42892116a6203371b63dbcf4e84ac9042/src/components/Navigation/Pagination/hooks/usePagination.ts#L120

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/hooks/usePagination.ts:29]: ScreenRecording2024-06-10at18 40 01-ezgif com-video-to-gif-converter

Comment (mainsmirnov) [reply src/components/Navigation/Pagination/hooks/usePagination.ts:29]: Right thing to do is remove onClick from the ellipse div

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/hooks/usePagination.ts:29]: Are you sure? I think someone might want to add some functionality on ellipsis click like in [antd](https://ant.design/components/pagination).

Comment (mainsmirnov) [reply src/components/Navigation/Pagination/hooks/usePagination.ts]: You can't change children of this parent component, so developer outside can't control this buttons

File: src/components/Navigation/Pagination/Pagination.module.css [hunk 2]
```diff
@@ -5,7 +5,7 @@
 }
 
 .wrapper--disabled {
-  opacity: .35;
```
Comment (mainsmirnov) [inline src/components/Navigation/Pagination/Pagination.module.css:8]: Why would you do that?

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/Pagination.module.css:8]: npx prettier did that

File: src/components/Navigation/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -74,7 +76,10 @@ export const Pagination = ({
       {...restProps}
     >
       {paginationItems.map((item) => {
-        const isEllipsis = [PaginationType.StartEllipsis, PaginationType.EndEllipsis].includes(item.type);
```
Comment (mainsmirnov) [inline src/components/Navigation/Pagination/Pagination.tsx:77]: Is it prettier auto change?

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/Pagination.tsx:77]: Yes

File: src/components/Navigation/Pagination/Pagination.module.css (hunk 2 above)
Comment (mainsmirnov) [reply src/components/Navigation/Pagination/Pagination.module.css:8]: I think it's wise to fix prettier in another PR and review it rules. This change conflicts with project codestyle

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/Pagination.module.css]: I returned initial prettier config and css changes of prettier. Also created issue about it https://github.com/Telegram-Mini-Apps/TelegramUI/issues/35

Comment (mainsmirnov) [reply src/components/Navigation/Pagination/Pagination.module.css]: This PR still contains unnecessary changes, like file renaming and lines splitting

Comment (bshelkhonov) [reply src/components/Navigation/Pagination/Pagination.module.css]: Right, fixed that.

**Comment by mainsmirnov** [comment]
Thank you!

**Pull request merged**